game/hud/src/components/AnchorIndicator/index.tsx

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import {Position} from '../../services/session/layout';
+
+export interface AnchorIndicatorProps {
+  name: string;     // widget name
+  pos: Position;    // current widget position
+  setAnchor: (name: string, pos: Position) => void;
+}
+
+export interface AnchorIndicatorState {
+}
+
+class AnchorIndicator extends React.Component<AnchorIndicatorProps, AnchorIndicatorState> {
+
+  constructor(props: AnchorIndicatorProps) {
+    super(props);
+  }
+
+  private anchorHere = (event: React.MouseEvent) => {
+    const el: HTMLDivElement = event.currentTarget as HTMLDivElement;
+    function a2n(a: string) : number {
+      switch(a) {
+        case 's': return -1;
+        case 'e': return 1;
```

Comment:
I actually don't particularly like the way this is done, it feels hacky.  I guess the real answer is to make these dots a component, and pass the position of the anchor as a property.